Mathlib/CategoryTheory/ComposableArrows.lean

```diff
@@ -0,0 +1,143 @@
+/-
+Copyright (c) 2023 Joël Riou. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joël Riou
+-/
+import Mathlib.AlgebraicTopology.Nerve
+
+/-!
+# Composable arrows
+
+If `C` is a category, the type of `n`-simplices in the nerve of `C` identifies
+to the type of functors `Fin (n + 1) ⥤ C`, which can be thought as families of `n` composable
+arrows in `C`. In this file, we introduce and study this category `ComposableArrows C n`
+of `n` composable arrows in `C`.
+
+TODO (@joelriou):
+* define various constructors for objects, morphisms, isomorphisms in `ComposableArrows C n`
+* extensionality lemmas for morphisms
+* extensionality lemmas for objects (useful for getting equalities of simplices in `nerve C`)
+* construction of `precomp F f : ComposableArrows C (n + 1)` when `F : ComposableArrows C n`
+and `f : X ⟶ F.left` with good definitional properties.
+* constructors like `mk₃ f g h : ComposableArrows C 3` which would take as inputs a certain
+number of composable morphisms
+* redefine `Arrow C` as `ComposableArrow C 1`?
+* construct some elements in `ComposableArrows m (Fin (n + 1))` for small `n`
+the precomposition with which shall induce funtors
+`ComposableArrows C n ⥤ ComposableArrows C m` which correspond to simplicial operations
+(specifically faces) with good definitional properties (this might be necessary for
+up to `n = 7` in order to formalize spectral sequences following Verdier)
+
+-/
+
+namespace CategoryTheory
+
+open Category
+
+variable (C : Type*) [Category C]
+
+/-- `ComposableArrows C n` is the type of functors `Fin (n + 1) ⥤ C`. -/
+abbrev ComposableArrows (n : ℕ) := Fin (n + 1) ⥤ C
+
+/-- The type of `n`-simplices in the simplicial set `nerve C` is
+definitionaly equal to `ComposableArrows C n`. -/
+example (n : ℕ) :
```

Comment:
I can imagine it might occasionally be useful to have this equality available for rewriting. Should it just get a name?

Reply:
I have made this a lemma.